plugins/modules/ndo_ipsla_track_list.py

```diff
@@ -0,0 +1,536 @@
+                        ),
+                    ),
+                    scope_uuid=dict(type="str"),
+                    scope=dict(
+                        type="dict",
+                        options=dict(
+                            name=dict(type="str", required=True),
+                            template=dict(type="str", required=True),
+                            schema=dict(type="str"),
+                        ),
+                    ),
+                    scope_type=dict(type="str", choices=["bd", "l3out"], required=True),
+                    weight=dict(type="int"),
+                ),
+                mutually_exclusive=[
+                    ("ipsla_monitoring_policy_uuid", "ipsla_monitoring_policy"),
+                    ("scope_uuid", "scope"),
+                ],
+                required_one_of=[
+                    ("ipsla_monitoring_policy_uuid", "ipsla_monitoring_policy"),
+                    ("scope_uuid", "scope"),
+                ],
+            ),
+            state=dict(type="str", choices=["absent", "query", "present"], default="present"),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["ipsla_track_list", "ipsla_track_list_uuid"], True],
+            ["state", "present", ["ipsla_track_list", "ipsla_track_list_uuid"], True],
+            ["state", "present", ["type"]],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    ipsla_track_list = module.params.get("ipsla_track_list")
+    description = module.params.get("description")
+    ipsla_track_list_uuid = module.params.get("ipsla_track_list_uuid")
+    type = module.params.get("type")
+    thresholds = {
+        "down": module.params.get("threshold_down"),
+        "up": module.params.get("threshold_up"),
+    }
+    members = module.params.get("members")
+    state = module.params.get("state")
+
+    # Validate
+    valid_upper = 100
+    valid_lower = 0
+    if type == "weight":
+        valid_upper = 255
+    for threshold_key, threshold_value in thresholds.items():
+        if threshold_value is not None and threshold_value not in range(valid_lower, valid_upper):
+            mso.fail_json(
+                msg="Invalid value provided for threshold_{0}: {1}; it must be in the range {2} - {3}".format(
+                    threshold_key, threshold_value, valid_lower, valid_upper
+                )
+            )
+
+    if thresholds["down"] is not None and thresholds["up"] is not None:
+        if thresholds["down"] > thresholds["up"]:
+            mso.fail_json(
+                msg="Invalid value provided for threshold_down: {0}; it must be less than or equal to threshold_up: {1}".format(
+                    thresholds["down"], thresholds["up"]
+                )
+            )
+
+    ops = []
+    match = None
+    obj_cache = {}
```

Comment:
The `obj_cache` is used so each `member` does not need to send an API call to get schema/template details multiple times.

Reply:
I think its a good idea for a future enhancement. We could potentially explore different libraries that may help. eg https://pypi.org/project/cachetools/
However, it would be quite a bit of work. It may also only be useful on modules with lists of dictionaries that need to call APIs for each item, like this module and `mso_schema_template_anp_epg_contract`. A simple dictionary cache like this is straight forward enough to implement on a case-by-case basis for now.